sys/stdio_uart/stdio_uart.c

```diff
@@ -45,16 +47,31 @@ extern ethos_t ethos;
 #define ENABLE_DEBUG 0
 #include "debug.h"
 
-
+#ifdef MODULE_STDIO_UART_RX
 static char _rx_buf_mem[STDIO_UART_RX_BUFSIZE];
 isrpipe_t stdio_uart_isrpipe = ISRPIPE_INIT(_rx_buf_mem);
+#endif
 
 void stdio_init(void)
 {
+    uart_rx_cb_t cb;
+    void *arg;
+
+#ifdef MODULE_STDIO_UART_RX
+    cb = (uart_rx_cb_t) isrpipe_write_one;
+    arg = &stdio_uart_isrpipe;
+#else
+#ifdef USE_ETHOS_FOR_STDIO
```

Comment:
I'm wondering if we could make stdio_uart_rx a dependency of ethos in the build system instead of preprocessor macros ?

Reply:
I am not sure about this... @kaspar030 any opinion?

Reply:
hm, ethos itself doesn't need *stdio*_uart_rx.
stdio-over-ethos should probably be a pseudomodule...

I guess as is this is fine for now.